### ISSUE TITLE
hoodie.remote.push creates conflicts

### ISSUE THREAD

**Issue by gr2m** [hoodiehq/hoodie#206]
If `hoodie.remote.push` gets executed multiple times, without the changes being pulled from remote (updating the _rev numbers), then the same objects will be pushed, but with different `_rev` ids.

To reproduce, run the following code, and she the _rev ids of the objects being pushed.

``` js
Backbone.hoodie.store.findAll().then(function(objects) {
  Backbone.hoodie.remote.push(objects);
  Backbone.hoodie.remote.push(objects);
})
```

To solve that and similar issues, I'd really like to build upon the great work of PouchDB. But until then, for this particular issue, I'd suggest we simply set a `_next_rev` property on each save, unless it is already set. That we we don't need to generate new `rev` numbers in the `hoodie.remote.push` method, but can use the ones that are set in `_next_rev` instead.

cc @janl @svnlto

**Comment by gr2m** [comment]
Multiple push might for example be caused by this line: https://github.com/hoodiehq/hoodie.js/blob/master/src/hoodie/account_remote.js#L107

I'd really like to have a pub/sub API that would assure that I only subscribe once to a specific event, even when the line above gets executed 10 times, I wont `remote.push` to be called only once when the `store:idle` event gets triggered.

**Comment by svnlto** [comment]
>  even when the line above gets executed 10 times

you'd want to implement your own 'debounce' method I guess.

**Comment by gr2m** [comment]
that'd be a possible workaround. But I think there are pub/sub systems that can prevent that from happening in the first place

**Comment by svnlto** [comment]
> that'd be a possible workaround. But I think there are pub/sub systems that can prevent that from happening in the first place

that'd wouldn't be  classic pub/sub tho.

**Comment by gr2m** [comment]
yeah you might be right, I'll open a separate issue for the pub/sub discussion.
What do you think about 4098c64b0a7a7976ef5d77a34209a305a61968eb? It's a simple hack, for a complex problem. I'll add tests, but wanted to discuss first

**Comment by gr2m** [comment]
fyi I've pushed the update to my hoodie app, and it works as a workaround to prevent conflicts from happening.
Also, the reason why this lead to so many conflicts was that the `POST /_bulk_docs` requests where pending for some time, due to heavy load.

**Comment by janl** [comment]
PouchDB will fix this.
